source/PlayerInfoPanel.cpp

```diff
@@ -651,15 +781,46 @@ void PlayerInfoPanel::DrawFleet(const Rectangle &bounds)
 }
 
 
+// Sorts the player's fleet given a comparator function (based on column)
+vector<shared_ptr<Ship>> PlayerInfoPanel::SortShips(bool sortDescending, ShipComparator &shipComparator)
+{
+	// Move flagship to first position
+	for(auto it = ships.begin(); it != ships.end(); ++it)
+		if(it->get() == player.Flagship())
+		{
+			iter_swap(it, ships.begin());
+			break;
+		}
+
+	std::stable_sort(ships.begin() + 1, ships.end(), [&](const shared_ptr<Ship> &lhs, const  shared_ptr<Ship> &rhs) {
+		if(sortDescending == true)
```

Comment:
If we don't make use of `std:: reverse`, note that you can equate the comparator result with the sort direction flag, rather than branch.